processing/src/main/java/org/apache/druid/query/scan/ScanQuery.java

```diff
@@ -36,26 +40,88 @@
 
 public class ScanQuery extends BaseQuery<ScanResultValue>
 {
-  public static final String RESULT_FORMAT_LIST = "list";
-  public static final String RESULT_FORMAT_COMPACTED_LIST = "compactedList";
-  public static final String RESULT_FORMAT_VALUE_VECTOR = "valueVector";
+  public enum ResultFormat
+  {
+    RESULT_FORMAT_LIST,
+    RESULT_FORMAT_COMPACTED_LIST,
+    RESULT_FORMAT_VALUE_VECTOR;
+
+    @JsonValue
+    @Override
+    public String toString()
+    {
+      switch (this) {
+        case RESULT_FORMAT_LIST:
+          return "list";
+        case RESULT_FORMAT_COMPACTED_LIST:
+          return "compactedList";
+        case RESULT_FORMAT_VALUE_VECTOR:
+          return "valueVector";
+        default:
+          return "";
+      }
+    }
+
+    @JsonCreator
+    public static ResultFormat fromString(String name)
+    {
+      switch (name) {
+        case "compactedList":
+          return RESULT_FORMAT_COMPACTED_LIST;
+        case "valueVector":
+          return RESULT_FORMAT_VALUE_VECTOR;
+        case "list":
+          return RESULT_FORMAT_LIST;
+        default:
+          throw new UOE("Scan query result format [%s] is not supported.", name);
+      }
+    }
+  }
+
+  public enum TimeOrder
```

Comment:
We could reuse the `OrderByColumnSpec.Direction` enum here, using `null` to represent the default no ordering case

Reply:
I'm fine with that.  However, OrderByColumnSpec.Direction defaults to null if the provided String value is neither ascending nor descending.   Is it good UX to have invalid input defaulting to null/NONE?